src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1359,131 +1360,200 @@ def update_dns_record_set(instance, metadata=None):
-                    else {'txt': ' '.join(record.value)} if record_type == 'txt' \
-                    else None
-                record_obj['name'] = record_set.name
-                record_obj['ttl'] = record_set.ttl
-                zone_obj[record_type].append(record_obj)
-
-    if 'soa' in zone_obj:
-        # there is only 1 soa record allowed, so it shouldn't be a list, take the first element
-        zone_obj['soa'] = zone_obj['soa'][0]
-        zone_obj['soa']['mname'] = zone_obj['soa']['mname'].rstrip('.') + '.'
-        zone_obj['soa']['rname'] = zone_obj['soa']['rname'].rstrip('.') + '.'
+        record_type = record_set.type.rsplit('/', 1)[1].lower()
+        record_set_name = record_set.name
+        record_data = getattr(record_set, _type_to_property_name(record_type), None)
+
+        # ignore empty record sets
+        if not record_data:
+            continue
+
+        if not isinstance(record_data, list):
+            record_data = [record_data]
+
+        if record_set_name not in zone_obj:
+            zone_obj[record_set_name] = OrderedDict()
+
+        for record in record_data:
+
+            record_obj = {'ttl': record_set.ttl}
+
+            if record_type not in zone_obj[record_set_name]:
+                zone_obj[record_set_name][record_type] = []
+
+            if record_type == 'aaaa':
+                record_obj.update({'ip': record.ipv6_address})
+            elif record_type == 'a':
+                record_obj.update({'ip': record.ipv4_address})
+            elif record_type == 'cname':
+                record_obj.update({'alias': record.cname})
+            elif record_type == 'mx':
+                record_obj.update({'preference': record.preference, 'host': record.exchange})
+            elif record_type == 'ns':
+                record_obj.update({'host': record.nsdname})
+            elif record_type == 'ptr':
+                record_obj.update({'host': record.ptrdname})
+            elif record_type == 'soa':
+                record_obj.update({
+                    'mname': record.host.rstrip('.') + '.',
+                    'rname': record.email.rstrip('.') + '.',
+                    'serial': record.serial_number, 'refresh': record.refresh_time,
+                    'retry': record.retry_time, 'expire': record.expire_time,
+                    'minimum': record.minimum_ttl
+                })
+                zone_obj['$ttl'] = record.minimum_ttl
+            elif record_type == 'srv':
+                record_obj.update({'priority': record.priority, 'weight': record.weight,
+                                   'port': record.port, 'target': record.target})
+            elif record_type == 'txt':
+                record_obj.update({'txt': ' '.join(record.value)})
+
+            zone_obj[record_set_name][record_type].append(record_obj)
 
     zone_file_text = make_zone_file(zone_obj)
 
-    with open(file_name, 'w') as f:
-        f.write(zone_file_text)
+    if file_name:
```

Comment:
I originally made it optional so it would just print if you didn't specify one, but you are right that we don't do that for similar commands, so I just removed it.